rclpy/rclpy/parameter_service.py

```diff
@@ -64,28 +66,37 @@ def __init__(self, node):
         )
 
     def _describe_parameters_callback(self, request, response):
-        for name in request.names:
-            try:
-                descriptor = self._node.describe_parameter(name)
-            except ParameterNotDeclaredException:
-                descriptor = ParameterDescriptor()
-            response.descriptors.append(descriptor)
+        node = self._node_weak_ref()
+        if node is not None:
```

Comment:
If the node is not available anymore is it sufficient to return the unmodified `response` or should any kind of exception be raised instead?

Same below.

Reply:
I think it's ok to return the unmodified response (which is the current behavior).

Also, it's currently impossible that the callback would be called after the node gets destroyed (as just the node is storing a reference to the created services), so the weak reference should be never invalid.

Reply:
> it's currently impossible that the callback would be called after the node gets destroyed ... so the weak reference should be never invalid.

If that is an invariant why not assert that explicitly?

Reply:
Fixed in https://github.com/ros2/rclpy/pull/490/commits/d41fc53d10d39eafdc943e839ddec2d1a9d883ac.